test/EndToEnd/WorkloadTests.cs

```diff
@@ -0,0 +1,134 @@
+
+        private static void TestBlazorWasmWithWithWorkload(string extraBuildArgs="", [CallerMemberName] string callerName = "")
+        {
+            DirectoryInfo testDirectory = TestAssets.CreateTestDirectory(callingMethod: callerName);
+            WorkloadTestEnvironment env = PrepareTestEnvironment(testDirectory.FullName);
+
+            new DotnetWorkloadCommand(env.dotnetUnderTest)
+                .ExecuteWithCapturedOutput("install microsoft-net-sdk-blazorwebassembly-aot --skip-manifest-update")
+                .Should().Pass();
+
+            string publishArgs = $"publish {extraBuildArgs}";
+            new DotnetCommand(env.dotnetUnderTest)
+                 .WithEnvironmentVariable("PATH", env.pathValue)
+                 .WithWorkingDirectory(Path.GetDirectoryName(env.projectFile))
+                 .Execute(publishArgs)
+                 .Should().Pass();
+        }
+
+        private static WorkloadTestEnvironment PrepareTestEnvironment(string baseDirectory)
+        {
+            string origDotnetPath = Path.GetDirectoryName(RepoDirectoriesProvider.DotnetUnderTest);
+
+            string packsDir = Path.Combine(origDotnetPath, "packs", "Microsoft.NET.Runtime.WebAssembly.Sdk");
+            Directory.Exists(packsDir)
+                .Should().BeFalse($"{packsDir} should not exist. Shared 'dotnet' installation in artifacts, should not have any packs installed");
+
+            string dotnetDirectory = Path.Combine(baseDirectory, "dotnet");
+            DirectoryCopy(origDotnetPath, dotnetDirectory);
+
+            string dotnetUnderTest = Path.Combine(dotnetDirectory, "dotnet");
+            dotnetUnderTest += RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? ".exe" : "";
+
+            string projectDirectory = Path.Combine(baseDirectory, "project");
+            string projectFile = TestTemplateCreate("blazorwasm", projectDirectory);
+
+            // override PATH since razor rely on PATH to find dotnet
+            string pathValue = dotnetDirectory;
+            // WasmApp.targets needs /bin/sh
+            pathValue += RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "" : ":/bin";
+
+            return new WorkloadTestEnvironment(dotnetUnderTest, projectFile, pathValue);
+        }
+
+        // code from https://docs.microsoft.com/en-us/dotnet/standard/io/how-to-copy-directories
+        private static void DirectoryCopy(string sourceDirName, string destDirName, bool copySubDirs = true)
+        {
+            // Get the subdirectories for the specified directory.
+            DirectoryInfo dir = new(sourceDirName);
+
+            if (!dir.Exists)
+            {
+                throw new DirectoryNotFoundException(
+                    "Source directory does not exist or could not be found: "
+                    + sourceDirName);
+            }
+
+            DirectoryInfo[] dirs = dir.GetDirectories();
```

Comment:
This can be done on Line 124. Also you could consider using `EnumerateDirectories()`